monai/handlers/utils.py

```diff
@@ -85,27 +85,38 @@ def evenly_divisible_all_gather(data: torch.Tensor) -> torch.Tensor:
     return torch.cat([data[i * max_len : i * max_len + l, ...] for i, l in enumerate(all_lens)], dim=0)
 
 
-def string_list_all_gather(strings: List[str], delimiter: str = "\t") -> List[str]:
+def string_list_all_gather(strings: List[str]) -> List[str]:
     """
     Utility function for distributed data parallel to all gather a list of strings.
+    Note that if the item in `strings` is longer than 1024 chars, it will be truncated to 1024.
 
     Args:
         strings: a list of strings to all gather.
-        delimiter: use the delimiter to join the string list to be a long string,
-            then all gather across ranks and split to a list. default to "\t".
 
     """
-    if idist.get_world_size() <= 1:
+    world_size = idist.get_world_size()
+    if world_size <= 1:
         return strings
 
-    _joined = delimiter.join(strings)
+    result: List[List[str]] = [[] for _ in range(world_size)]
+    # get length of strings
+    length = len(strings)
+    all_lens = idist.all_gather(length)
```

Comment:
thanks, would be nice to have some filename like test cases

Reply:
Hi @wyli ,

We already have the tests to cover filenames all-gather in multi-gpu:
https://github.com/Project-MONAI/MONAI/blob/master/tests/test_handler_metrics_saver_dist.py#L89

Thanks.

Reply:
the existing tests can't detect bug #1634

Reply:
OK, do you mean to add a test that every single string in the list is longer than 1024 or just the total length of the list > 1024 as the bug?

Thanks.

Reply:
thanks I'm more interested in the latter case but I think you can test both

Reply:
Add test case for the later.

Reply:
Added test cases to cover both.